# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasRenderer.cpp
# Commit: fd4a74ec97a5
# Full Hash: fd4a74ec97a5d9540157f30dfc6184031c2499ac
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 4cda148ef28f -r fd4a74ec97a5 gfx/layers/CanvasRenderer.cpp
--- a/gfx/layers/CanvasRenderer.cpp	Mon Jun 15 18:26:00 2020 +0000
+++ b/gfx/layers/CanvasRenderer.cpp	Mon Jun 15 18:26:02 2020 +0000
@@ -46,7 +46,8 @@
   return mData.GetContext() == aData.GetContext();
 }
 
-std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot() const {
+std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot(
+    const bool requireAlphaPremult) const {
   const auto context = mData.GetContext();
   if (!context) return nullptr;
   const auto& provider = context->GetBufferProvider();
@@ -57,7 +58,7 @@
     ss = provider->BorrowSnapshot();
   }
   if (!ss) {
-    ss = context->GetFrontBufferSnapshot();
+    ss = context->GetFrontBufferSnapshot(requireAlphaPremult);
   }
   if (!ss) return nullptr;
 